lib/mysql.js

```diff
@@ -200,17 +200,8 @@ MySQL.prototype.executeSQL = function(sql, params, options, callback) {
   }
 };
 
-/**
- * Update if the model instance exists with the same id or create a new instance
- *
- * @param {String} model The model name
- * @param {Object} data The model instance data
- * @param {Function} [callback] The callback function
- */
-MySQL.prototype.updateOrCreate = MySQL.prototype.save =
-  function(model, data, options, callback) {
-    var fields = this.buildFields(model, data);
-
+ MySQL.prototype.modifyOrCreate = function(model, data, options, buildFields ,callback) {
+    var fields = buildFields(model, data);
```

Comment:
You probably understand the impl more than I do since you've had previous discussions with @bajtos. However, instead of passing in the function, can't you just pass in the fields?

MySQL.prototype.modifyOrCreate = function(model, data, options, fields, cb) {
  ...   
});
```

Then in updateOrCreate:

```
MySQL.prototype.updateOrCreate = function(model, data, options, callback) {
  var fields = this.buildReplaceFields(model, data);
  this.modifyOrCreate(model, data, options, fields, callback);
});
```

and in replace:

```
MySQL.prototype.replaceOrCreate = function (model, data, options, cb) {
  var fields = this.buildFields(model, data);
  this.modifyOrCreate(model, data, options, fields, callback);  
});
```

@raymondfeng Thoughts? ^